src/modules/SdnDiag.NetworkController/public/Get-SdnSlbStateInformation.ps1

```diff
@@ -17,23 +20,26 @@ function Get-SdnSlbStateInformation {
     .EXAMPLE
         Get-SdnSlbStateInformation -NcUri "https://nc.contoso.com" -VirtualIPAddress 41.40.40.1
     .EXAMPLE
-        Get-SdnSlbStateInformation -NcUri "https://nc.contoso.com" -Credential (Get-Credential)
+        Get-SdnSlbStateInformation -NcUri "https://nc.contoso.com" -NcRestCredential (Get-Credential)
     .EXAMPLE
         Get-SdnSlbStateInformation -NcUri "https://nc.contoso.com" -ExecutionTimeout 1200
     #>
 
-    [CmdletBinding()]
+    [CmdletBinding(DefaultParameterSetName = 'RestCredential')]
     param (
         [Parameter(Mandatory = $true)]
         [uri]$NcUri,
 
         [Parameter(Mandatory = $false)]
         [IPAddress]$VirtualIPAddress,
 
-        [Parameter(Mandatory = $false)]
+        [Parameter(Mandatory = $false, ParameterSetName = 'RestCredential')]
         [System.Management.Automation.PSCredential]
         [System.Management.Automation.Credential()]
-        $Credential = [System.Management.Automation.PSCredential]::Empty,
+        $NcRestCredential = [System.Management.Automation.PSCredential]::Empty,
+
+        [Parameter(Mandatory = $true, ParameterSetName = 'RestCertificate')]
```

Comment:
Review `Mandatory` attribute

Reply:
This is correct when using ParameterSetName and helps define what is required. It enables unique properties to be required depending on the ParameterSetName that is used.